CMakeLists.txt

```diff
@@ -100,6 +100,16 @@ option(USE_BLOSC [=[
 Use blosc while building openvdb components. If OPENVDB_BUILD_CORE is OFF, CMake attempts to query the located
 openvdb build configuration to decide on blosc support. You may set this to on to force blosc to be used if you
 know it to be required.]=] ON)
+option(USE_JEMALLOC [=[
+Use Jemalloc while building openvdb components. If USE_JEMALLOC is OFF, CMake attempts to query the
+located openvdb build configuration to decide on Jemalloc support. You may set this to off to disable Jemalloc.
+Although not strictly required, it is strongly recommended to use a concurrent memory allocator.
+Note that you must choose between Jemalloc or TBB malloc, or otherwise disable them both.]=] ON)
```

Comment:
This is on by default, is that correct? is jemalloc usually available on mac/centos/windows outside of houdini/maya builds?

Reply:
FWIW, Maya ships with both tbbmalloc and (as of Maya 2017) "awjemalloc", which seems to be jemalloc with renamed symbols.  Since we already require TBB, it might be better to make tbbmalloc the default allocator outside of Houdini builds.

Reply:
I have found jemalloc to provide a decent performance improvement over tbbmalloc, so perhaps we make jemalloc the default outside of non-Maya builds?

Reply:
@Idclip - do you have a good idea how we would do this in the CMake? Make USE_TBBMALLOC default to ON and USE_JEMALLOC default to OFF when USE_MAYA is ON otherwise OFF and ON respectively?

Reply:
Does CMake support enum options?  jemalloc and tbbmalloc should be mutually exclusive, so what we probably really want is a CONCURRENT_MALLOC option for which the choices are NONE, JEMALLOC, TBBMALLOC, and APPLICATION, where the latter specifies jemalloc for Houdini builds and tbbmalloc for Maya builds.

Reply:
Hmm. On the one hand, I think I prefer OFF/ON options because then the common use case of enabling one (USE_JEMALLOC=ON) fits in better with how the other variables work.

The contrary argument is that if we were to add support for another memory allocator, this gets pretty fiddly to support, so I suppose the common use case should be *not* changing the default at all.

I think your suggestion is best, except perhaps using "AUTO" instead of "APPLICATION" which can imply jemalloc when building outside of Houdini and Maya.

Reply:
> Does CMake support enum options? 
afaik no - you have to use strings i.e.

```
if(NOT CMAKE_BUILD_TYPE)
  set(CMAKE_BUILD_TYPE Release CACHE STRING
    "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel." FORCE
  )
endif()
```
You can have dependent CMake options - we use them a bit:

```
CMAKE_DEPENDENT_option(USE_JEMALLOC "use jemalloc" ON "USE_MAYA" OFF)
CMAKE_DEPENDENT_option(USE_TBBMALLOC "use tbbmalloc" OFF "USE_MAYA" ON)
```

https://cmake.org/cmake/help/latest/module/CMakeDependentOption.html

Reply:
For AUTO I would say the logic should be jemalloc for Houdini builds; tbbmalloc for Maya builds; and for all other builds jemalloc if it is available, otherwise tbbmalloc.

Reply:
I agree that's ideal, but I'm not sure how easy it is to implement the "if it is available" part.

Reply:
You can remove the REQUIRED keyword from the find_package call so that it doesn't error when searching for jemalloc. Then you can use if(TARGET Jemalloc::jemalloc) ... To check to see if the imported target was created/found

Reply:
Thanks for the suggestion, I'll give that a go.